portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,29 +59,89 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    int maxComments = getMaxCommentsToReturn(request);
-
+    JsonObject json = new JsonObject();
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      String name = getUserName(userService.getCurrentUser().getUserId());
+
+      // If the user has logged in for the first time, set their name as their Google account nickname.
+      if (name == null) {
```

Comment:
If the user updates their name, how would this be handled? I'm wondering if we need to store our own id->name mapping and managing user names. UserService may have their own privacy requirements and it may not be clear to users also that this website would be storing a copy of the name info in its own datastore.

Reply:
Ah, as I read more of the code I see there's a way to update your stored name. You are just seeding the name from the UserService. I was curious about what happens if they updated their nickname on the UserService. I'm still curious if we want to maintain different identities?

Reply:
My intention was to allow users to have a display name specific to the portfolio and just use the nickname from UserService as an initial value before the users update it. So if the nickname in UserService changes after a user first logs in, it should not have any effect. Does that answer your question?